textpattern/vendors/Textpattern/Tag/Registry.php

```diff
@@ -74,11 +74,9 @@ public function register($callback, $tag = null)
      * @return string|null The tag's results
      */
 
-    public function process($tag, array $atts = null, $thing = null)
+    public function process($tag, array $atts = array(), $thing = null)
     {
-        if ($this->isRegistered($tag)) {
```

Comment:
As I said in commit 35d62a4:
Don't check ifRegistered twice. It's already done in txplib_publish.php before calling this process() function. I was tempted to leave it in here and check if the return value was 'null' in txplib_publish.php, but some taghandlers (txp:variable!) don't return a value (=null) so there wouldn't be any way to differentiate it from a non-registered tag.